src/main/java/com/rybka/service/ApiConnectorService.java

```diff
@@ -0,0 +1,36 @@
+package com.rybka.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rybka.CurrencyAPIConstants;
+import com.rybka.exception.CurrencyAPICallException;
+import com.rybka.model.ExchangeResponse;
+import lombok.extern.log4j.Log4j;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+@Log4j
+public class ApiConnectorService {
+
+    private final HttpClient client = HttpClient.newHttpClient();
```

Comment:
It's not a good practice to provide instances directly in the class. 

1. Imagine we have another service that uses HttpClient as well. In result, we have to create  another one client in that service. More services more httpClients. The same for ObjectMapper. Why do we need several instances if we can have just one. 

2. Testing purpose. In case, we want to create test for our ApiConnectorService. Tested method  `retrieveRates` will make a real call to the service, it could be fine for Integration test but not for unit test. In the unit test we use mockers for depended services(httpClient, objectMapper) to provide mocked calls/results. Read more about testing. 

3. DI. The best practice in the Spring IoC container is to provide dependencies(beans) by constructors. One of the next tasks will be upgrading application with Spring Core. You'll see benefits there ;)

Reply:
fixed